src/Compiler/Mapper/Object/MapDateTimeImmutable.php

```diff
@@ -5,28 +5,51 @@
 use Attribute;
 use DateTimeImmutable;
 use DateTimeInterface;
+use DateTimeZone;
 use PhpParser\Node\Expr;
 use PHPStan\PhpDocParser\Ast\Type\IdentifierTypeNode;
 use PHPStan\PhpDocParser\Ast\Type\TypeNode;
 use ShipMonk\InputMapper\Compiler\CompiledExpr;
 use ShipMonk\InputMapper\Compiler\Mapper\MapperCompiler;
 use ShipMonk\InputMapper\Compiler\Php\PhpCodeBuilder;
 use ShipMonk\InputMapper\Runtime\Exception\MappingFailedException;
+use function count;
+use function is_array;
+use function is_string;
 
 #[Attribute(Attribute::TARGET_PARAMETER)]
 class MapDateTimeImmutable implements MapperCompiler
 {
 
+    /**
+     * @param string|non-empty-list<string> $format
+     */
     public function __construct(
-        public readonly string $format = DateTimeInterface::RFC3339,
+        public readonly string|array $format = [DateTimeInterface::RFC3339, DateTimeInterface::RFC3339_EXTENDED],
         public readonly string $formatDescription = 'date-time string in RFC 3339 format',
+        public readonly ?string $timezone = null,
```

Comment:
alternative to single `$timezone` parameter would be to have two distinct parameters for each purpose, i.e. 

* `$defaultTimezone` for `createFromFormat()` and
* `$targetTimezone` for `setTimezone()`

Reply:
I like it, but I¨d document it in phpdoc. Maybe even add, what does it mean when `$format` is array, I was not sure till I saw implementation.